colossus/src/main/scala/colossus/protocols/http/HttpResponseParser.scala

```diff
@@ -19,12 +19,15 @@ object HttpResponseParser {
   def apply(size: DataSize = DefaultMaxSize) = maxSize(size, response)
 
   def response: Parser[HttpResponse] = firstLine ~ headers |> {case (version, code) ~ headers =>
-    val clength = headers.collectFirst{case (c,v) if (c == HttpHeaders.ContentLength) => v.toInt}.getOrElse(0)
-    if (clength > 0) {
-      bytes(clength) >> {body => HttpResponse(version, code, body, headers)}
-    } else {
-      const(HttpResponse(version, code, ByteString(), headers))
-    }
+    val clength = headers.collectFirst{case (c,v) if (c == HttpHeaders.ContentLength) => v.toInt}
+    val encoding = headers.collectFirst{case (c,v) if (c == HttpHeaders.TransferEncoding) => v.toLowerCase}
```

Comment:
Same remark as with the request parser with regards to content length and chunk encoded transfer headers both being present.